### DISCUSSION TITLE
How to specify multiple xkb_options?

### DISCUSSION THREAD

**Discussion by probonopd** [WayfireWM/wayfire#2030] [Q&A]
How to specify multiple xkb_options?

I would like to swap the left and right Ctrl and Alt keys, like this:

```
[input]
xkb_options=ctrl:swap_lalt_lctl,ctrl:swap_ralt_rctl
```

But apparently only `ctrl:swap_lalt_lctl` gets applied.

I also tried 

```
xkb_options=ctrl:swap_lalt_lctl+ctrl:swap_ralt_rctl
```

and 

```
xkb_options=ctrl:swap_lalt_lctl ctrl:swap_ralt_rctl
```

**Comment by soreau** [comment]
Wayfire passes the options to [xkb_map_new_from_names()](https://github.com/WayfireWM/wayfire/blob/master/src/core/seat/keyboard.cpp#L149), the [documentation for which](https://xkbcommon.org/doc/current/structxkb__rule__names.html#a556899eafb333c440be64ede408644df) shows you are correct in assuming it should be a comma-separated list. I'm wondering if switching the options, the first one will work in the list or the second. There are a couple of things to try though, try putting the comma separated list in quotations, `setxkbmap` at runtime or setting `XKB_DEFAULT_OPTIONS` before starting wayfire.

**Reply by probonopd** [reply]
The first option works, the second is ignored.

**Reply by probonopd** [reply]
`xkb_options="ctrl:swap_lalt_lctl,ctrl:swap_ralt_rctl"` does not work at all.
